From 301ade3010464272270a1fcc15f152e216500087 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 31 Oct 2018 16:54:41 +0800
Subject: [PATCH 4967/5242] MLK-19750-5: ARM64: dts: workaround 192kHz
 recording noise issue

commit  e818f109021f1257635393b5849b669bb47a52bf from
https://source.codeaurora.org/external/imx/linux-imx.git

There is noise issue with 192kHz recoding with ESAI + CS42xx8 on imx8qxp
mek board.

This issue is caused by the round trip delay due to longer trace length
on board. After we switch to tx master, rx slave mode, the issue is gone.
so the setting can workaround the issue, the reason is that the bitclock,
frame clock and data is generated from one side, for recording, is from
codec, the asynchronous of clock and data is eliminated.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-mek-m4.dts      |    1 +
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
index 69aba7e..63c9a00 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
@@ -40,6 +40,7 @@
 		VLS-supply = <&reg_audio>;
 		VLC-supply = <&reg_audio>;
 		reset-gpio = <&pca9557_b 1 1>;
+		fsl,txs-rxm;
 		status = "okay";
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 51399a7..e8a3526 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -228,6 +228,7 @@
 			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>;
 	assigned-clock-parents = <&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_CLK>;
 	assigned-clock-rates = <0>, <786432000>, <49152000>, <12288000>, <49152000>;
+	fsl,txm-rxs;
 	status = "okay";
 };
 
@@ -772,6 +773,7 @@
 				<&clk IMX8QXP_AUD_ACM_AUD_REC_CLK0_DIV>,
 				<&clk IMX8QXP_AUD_MCLKOUT0>;
 		assigned-clock-rates = <786432000>, <49152000>, <12288000>, <12288000>;
+		fsl,txs-rxm;
 		status = "okay";
 	};
 };
-- 
1.7.9.5

